packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -485,45 +487,43 @@ class TimeInput extends React.Component {
     if ((event.key === '-' || event.key === '_') && !this.props.atMinDate) {
       const currentTimeValue = this.formatHour(hour, meridiem).concat(':', minute).concat(this.props.showSeconds ? ':'.concat(second) : '');
       if (TimeUtil.validateTime(currentTimeValue, this.props.showSeconds)) {
-        if (minute === '0' || minute === '00') {
-          minute = '59';
-          if (hour === '0' || hour === '00') {
-            hour = '23';
-          } else {
-            if (variant === TimeUtil.FORMAT_12_HOUR && hour === '12') {
-              if (meridiem === this.anteMeridiem) {
-                meridiem = this.postMeridiem;
-              } else {
-                meridiem = this.anteMeridiem;
-              }
+        if (inputType === 3 && this.props.showSeconds) {
+          if ((second === TimeUtil.SingleZeroDigit || second === TimeUtil.DoubleZeroDigit)) {
```

Comment:
Nit : ```suggestion
          if (TimeUtil.initialValue.includes(second)) {
```
singleZero and doubleZeroDigit can be added to single array !!

Reply:
Updated - https://github.com/cerner/terra-framework/pull/1966/commits/4f5861d144599c94974d061f9a0d9d39d5c0f30b